packages/babel-plugin-transform-classes/src/vanilla.js

```diff
@@ -50,14 +34,20 @@ const verifyConstructorVisitor = visitors.merge([
     },
 
     ThisExpression(path) {
-      if (this.isDerived && !this.hasBareSuper) {
-        const fn = path.find(p => p.isFunction());
-
-        if (!fn || !fn.isArrowFunctionExpression()) {
-          throw path.buildCodeFrameError(
-            "'this' is not allowed before super()",
-          );
+      if (this.isDerived) {
+        if (path.parentPath.isMemberExpression({ object: path.node })) {
+          // In cases like this.foo or this[foo], there is no need to ass
```

Comment:
Super small nit: extraneous `ass` :)

Reply:
Lol 🤣 
I will correct it later and then merge the pr.

Reply:
@nicolo-ribaudo lol sounds good